api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -114,6 +114,9 @@ message OAuth2Config {
   // Defines how ``client_id`` and ``client_secret`` are sent in OAuth client to OAuth server requests.
   // RFC https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1
   AuthType auth_type = 11 [(validate.rules).enum = {defined_only: true}];
+
+  // Attempt to update access token using by refresh token if that is provided by authorization_endpoint.
+  bool use_refresh_token = 12;
```

Comment:
Should this default to true? If so use BoolValue and document that.

Reply:
The default is false. Because I don't consider to change old behaviour after update envoy. If user wants to use refresh token then manually set the flag to true.

Reply:
I see your point of not breaking current behavior, and I agree it should be false by default at the moment.
It may be better to use a BoolValue type, as in the future it may make sense to convert the default value to true (see https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md#api-style-guidelines re "wrapped scalar types").

Reply:
I don't want to make this changes, because I don't have more time to this PR. I killed very much time to fix minor changes. Sorry! I think it is easy to setup exactly refresh token support.

Reply:
Can you please elaborate what you mean by "it is easy to exactly set a refresh token support"?

Reply:
I mean user can set true or false clearly on configuration if user knows whether or not the oauth server supports refresh token.

Reply:
> The default is false. Because. I don't consider to change automatically old behaviour after update envoy. If user wants to use refresh token then manually set the flag to true. In the future we can change default behaviour

If in the future we want to be able to update the default value of this field, it should be a wrapped type, rather than a primitive.
/wait-any

Reply:
I have changed type of veriable to BoolValue as you requested